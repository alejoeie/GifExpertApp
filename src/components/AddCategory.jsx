import { useState } from "react"
import PropTypes from 'prop-types'

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setinputValue] = useState('');
    
    const onInputChange = (event) => {
        console.log(event.target.value);
        setinputValue(event.target.value);
    }

    const onSubmit = (event) => {
        console.log("Hello world from the very submit");
        event.preventDefault();
        if (inputValue.trim().length <=1 ) return;
        onNewCategory( inputValue.trim() );
        setinputValue('');
    }
    return (
        <form onSubmit={ (event) => {onSubmit(event)}} aria-label="form">
            <input 
            type="text"
            placeholder="Buscar gifs, ej: Dragon Ball"
            value={inputValue}
            onChange={ onInputChange }
            />
        </form>
    
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}