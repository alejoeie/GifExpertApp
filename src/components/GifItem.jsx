import React from 'react'
// import {p} from 'prop-types';
import PropTypes from 'prop-types'

export const GifItem = ({title, url, id}) => {
    // console.log(title, url, id);
  return (
    <div className='card'>
        <img src={url} alt={ title } id={id}/>
        <p>{title}</p>
    </div>
  )
}

// GifItem.
// GifIte
GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

// export default GifItem

