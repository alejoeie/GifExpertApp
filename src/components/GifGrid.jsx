import React from 'react'
import {GifItem} from './GifItem';
import useFetchGifs from '../hooks/useFetchGifs'
function GifGrid({category}) {

    const { images, isLoading } = useFetchGifs( category );


  return (
    <>  
        {/* <h3 className='text-3xl font-bold dark:text-white'>{category.toUpperCase()}</h3> */}
        {
            isLoading && (<h2>Cargando...</h2>)
        }
        
        <div className='card-grid'>
            
            {images.map( (image) => (
                <GifItem 
                    key={image.id}
                    {...image}
                />
            ))}
        </div>
        
    </>
  )
}

export default GifGrid