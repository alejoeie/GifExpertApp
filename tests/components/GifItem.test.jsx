import { render, screen } from "@testing-library/react"
import {GifItem} from "../../src/components/GifItem"

describe('Pruebas en GifItem', () => {
    const title = 'Tanjiro';
    const url  = 'https://tanjiro.com/'
    test('Must match snapshot', () => {

        const { container } = render(<GifItem title={title} url = {url} />);
        expect(container).toMatchSnapshot();
    });

    test('Should show image with URL and indicated ALT', () => {
        render(<GifItem title={title} url={url}/>)
        // expect( screen.getByRole('img').src).toBe( url );

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect( alt ).toBe(alt);
    }); 

    test("Should show the component title", () => {
        render(<GifItem title={title} url={url}/>);

        expect( screen.getByText( title ) ).toBeTruthy();
    })
})