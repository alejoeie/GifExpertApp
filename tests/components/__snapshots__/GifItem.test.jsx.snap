// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Must match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Tanjiro"
      src="https://tanjiro.com/"
    />
    <p>
      Tanjiro
    </p>
  </div>
</div>
`;
